### PR TITLE
treewide: align headers c++ guards fixes

### PR DESCRIPTION
**Pull request by eivindj-nordic**
* Make sure all C++ C includes have an end.
* Correct order of ifdefs and doxygen so that they are in the correct order. The order is not aligned from file to file, though it is corrected per file.
* Added space after license in some headers and source files.

### CODE REVIEW

**Review by krish2718 — APPROVED** [review]
For Wi-Fi parts (hostap, TWT sample)

**Review by rado17 — APPROVED** [review]

**Review by tomi-font — APPROVED** [review]
For SLM.

**Review by juhaylinen — APPROVED** [review]

**Review by joerchan — APPROVED** [review]

**Review by MarkusLassila — APPROVED** [review]

**Review by SeppoTakalo — APPROVED** [review]

**Review by ludvigsj — APPROVED** [review]
For Bluetooth Mesh

**Review by sigvartmh — APPROVED** [review]

File: include/caf/events/module_state_event.h [hunk 1]
```diff
@@ -4,16 +4,16 @@
...
-#define _MODULE_STATE_EVENT_H_
-
 /**
  * @file
  * @defgroup caf_module_state_event CAF Module State Event
  * @{
  * @brief CAF Module State Event.
  */
 
+#ifndef _MODULE_STATE_EVENT_H_
```
Comment (MarekPieta) [inline include/caf/events/module_state_event.h:14]: Please maintain consistent approach with other CAF events

File: include/sensor/pmw3360.h [hunk 2]
```diff
@@ -14,16 +15,16 @@
...
-extern "C" {
-#endif
-
 /**
  * @defgroup pmw3360 PMW3360 motion sensor driver
  * @{
  * @brief PMW3360 motion sensor driver.
  */
 
+#ifdef __cplusplus
```
Comment (MarekPieta) [inline include/sensor/pmw3360.h:24]: Please align the approach for this file with paw3212 header.

**Review by MarekPieta — CHANGES_REQUESTED** [review]
Two small fixes

File: include/caf/events/module_state_event.h (hunk 1 above)
Comment (eivindj-nordic) [reply include/caf/events/module_state_event.h]: updated.

File: include/sensor/pmw3360.h (hunk 2 above)
Comment (eivindj-nordic) [reply include/sensor/pmw3360.h]: updated

**Review by simensrostad — APPROVED** [review]

**Review by MarekPieta — APPROVED** [review]

**Review by lemrey — APPROVED** [review]

**Review by anangl — APPROVED** [review]

**Pull request merged**